src/qmqttclient.cpp

```diff
@@ -471,6 +477,29 @@ QString toString(const QList<QSslError> &sslErrors) {
     return sslErrorString;
 }
 
+
+/*!
+   \internal
+ */
+bool QMqttClientPrivate::sslErrorsAllowed(const QList<QSslError> &errors) const
+{
+    if (!m_allowedSslErrors.isEmpty())
+    {
+        QSet<QSslError> errorsSet;
+        for (const auto &error : errors)
+        {
+            errorsSet << QSslError(error.error());
+        }
```

Comment:
Why not use `QSet<QSslError>(errors.constBegin(), errors.constEnd())`?

Reply:
I have tested with your proposal "QSet<QSslError>(errors.constBegin(), errors.constEnd())" and added debug statements, but the substraction is not empty.
Output of debug statements: errorsSet QSet("The certificate is self-signed, and untrusted", "The host name did not match any of the valid hosts for this certificate")
m_allowedSslErrors QSet("The root certificate of the certificate chain is self-signed, and untrusted", "The certificate is self-signed, and untrusted", "The host name did not match any of the valid hosts for this certificate")
subtraction QSet("The certificate is self-signed, and untrusted", "The host name did not match any of the valid hosts for this certificate").
In our use case m_allowedSslErrors is defined using the QSslError constructor without a certificate QSslError::QSslError(QSslError::SslError error), e.g. QSet<QSslError>({QSslError::HostNameMismatch, QSslError::SelfSignedCertificate, QSslError::SelfSignedCertificateInChain})
The QSslErrors received with the QWebSocket::sslErrors signal are probably defined with the QSslError constructor with a certificate QSslError::QSslError(QSslError::SslError error, const QSslCertificate &certificate)
As the QSslErrors comparison operator compares both the error() and the certificate() the subtraction is not empty.
If I construct a QSet<QSslError> errorsSet using only the .error() value, the subtraction is empty.

Reply:
Aha! Good point. Maybe add some comment above to explain this situation?

Reply:
However, I think you must also make sure that the `m_allowedSslErrors` only contain errors with the certificate stripped off

Reply:
Also stripped off certificates of m_allowedErrors before subtraction.